src/borg/archive.py

```diff
@@ -195,6 +194,22 @@ def backup_io_iter(iterator):
                 return
         yield item
 
+@contextmanager
+def OsOpen(*, flags, path=None, parent_fd=None, name=None, noatime=False, op='open'):
+    with backup_io(op):
+        fd = os_open(path=path, parent_fd=parent_fd, name=name, flags=flags, noatime=noatime)
+    try:
+        yield fd
+    finally:
+        try:
+            os.close(fd)
+        except OSError as err:
+            if err.errno == errno.ENODEV and sys.platform == 'darwin':
+                # macOS raises this when we try to close a character device
+                pass
```

Comment:
looks like macOS not only has issues closing chr devices, but also opening blk devices.
maybe we need other open flags?

Reply:
solved by not opening devices any more.